hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/nativeio/NativeIO.java

```diff
@@ -620,7 +621,7 @@ private static String getName(IdCache domain, int id) throws IOException {
         ? USER_ID_NAME_CACHE : GROUP_ID_NAME_CACHE;
       String name;
       CachedName cachedName = idNameCache.get(id);
-      long now = System.currentTimeMillis();
+      long now = Time.monotonicNow();
       if (cachedName != null && (cachedName.timestamp + cacheTimeout) > now) {
```

Comment:
When switching from System.currentTimeMillis, or Time.now() to Time.monotonicNow(), keep in mind that the latter can return a negative number.   In this case, cachedName.timestamp could be negative due to overflow, while now might still be positive, in which case this code will not take the right path.  The correct comparison would be `((now - cachedName.timestamp) < cacheTimeout)`
But I would also avoid using monitonicNow for any field called `timestamp`.